e2e/components/autocomplete-e2e.spec.ts

```diff
@@ -0,0 +1,50 @@
+import {browser, by, element} from 'protractor';
+import {screenshot} from '../screenshot';
+import {expectToExist} from '../util/index';
+
+
+describe('autocomplete', () => {
+  const autocompletePanelSelector = '.mat-autocomplete-panel.mat-autocomplete-visible';
+  let page: AutocompletePage;
+
+  beforeEach(() => page = new AutocompletePage());
+
+  it('should open the panel when the input is focused', async () => {
+    expectToExist(autocompletePanelSelector, false);
+    page.autocompleteInput().click();
+
+    expectToExist(autocompletePanelSelector);
+    expect(await page.autocompletePanel().getText()).toEqual('One\nTwo\nThree\nFour');
+    screenshot();
+  });
+
+  it('should close the panel when an option is clicked', async () => {
+    page.autocompleteInput().click();
+    page.option('One').click();
+
+    expectToExist(autocompletePanelSelector, false);
+    screenshot();
+  });
+
+  it('should set the selected option text to the input', async () => {
+    page.autocompleteInput().click();
+    page.option('One').click();
+
+    expect(await page.getInputText()).toEqual('One');
+  });
+
+  it('should trigger filtering with proper key', async () => {
+    page.autocompleteInput().sendKeys('T');
+
+    expect(await page.autocompletePanel().getText()).toEqual('Two\nThree');
+    screenshot();
+  });
+});
+
+class AutocompletePage {
+  constructor() { browser.get('/autocomplete'); }
+  autocompleteInput = () => element(by.id('autocomplete-input'));
```

Comment:
Personally, I don't know that its strictly necessary to have a class to manage retrieving the elements as you have here.  If you would like to keep the class, rather than assigning these are attributes which are functions, can you use typical method declaration?


```ts
class AutocompletePage {
  constructor() {
    browser.get('/autocomplete');
  }

  autocompleteInput() {
    return element(by.id('autocomplete-input'));
  }
...
}
```

Reply:
Agreed. I don't think we have that many tests where setups like this start making sense.

Reply:
Personally, I prefer to user page objects, because it separates the test logic and elements selection. I've seen this pattern in `menu-e2e.spec.ts` and decided that it is acceptable to use it. I should remove this class, right?

Reply:
We haven't been super consistent about when we use page objects (IIRC only the menu tests have them). Let's keep it for now and we can decide later on whether we want to remove it or switch all of the other e2e tests to use page objects as well.